frontend/javascripts/oxalis/view/action-bar/private_links_view.tsx

```diff
@@ -0,0 +1,347 @@
+      items={[
+        {
+          label: "24 hours",
+          key: "24 hours",
+        },
+        {
+          label: "1 week",
+          key: "1 week",
+        },
+        {
+          label: "6 months",
+          key: "6 months",
+        },
+        {
+          label: "1 year",
+          key: "1 year",
+        },
+      ]}
+    />
+  );
+
+  if (linkItem.expirationDateTime == null) {
+    return (
+      <Dropdown overlay={expirationMenu}>
+        <Space style={{ color: "var(--ant-text-secondary)" }}>
+          Add Expiration Date
+          <DownOutlined />
+        </Space>
+      </Dropdown>
+    );
+  }
+
+  const maybeWarning =
+    Number(new Date()) > linkItem.expirationDateTime ? (
+      <Tooltip title="This link has expired">
+        <InfoCircleOutlined style={{ color: "var(--ant-error)" }} />
+      </Tooltip>
+    ) : null;
+
+  return (
+    <span>
+      <FormattedDate timestamp={linkItem.expirationDateTime} />
+      <Popover
+        content={
+          <>
+            <div>
+              <DatePicker
+                onChange={onChange}
+                // @ts-ignore
+                defaultValue={moment(linkItem.expirationDateTime)}
+                allowClear={false}
+              />
+            </div>
+            <Button
+              type="link"
+              onClick={() => updateMutation.mutate({ ...linkItem, expirationDateTime: null })}
+            >
+              Remove expiration date
+            </Button>
+          </>
+        }
+        title="Set an expiration date"
+        trigger="click"
+      >
+        <EditOutlined style={{ marginLeft: 4 }} />
+        {maybeWarning}
+      </Popover>
+    </span>
+  );
+}
+
+function PrivateLinksView({ annotationId }: { annotationId: string }) {
+  const { error, data: links } = useLinksQuery(annotationId);
+  const createLinkMutation = useCreateLinkMutation(annotationId);
+  const deleteMutation = useDeleteLinkMutation(annotationId);
+
+  if (error) {
+    return <span>Error while loading the private links: {error}</span>;
+  }
+
+  const columns: ColumnsType<ZarrPrivateLink> = [
+    {
+      title: "URL",
+      key: "name",
+      render: (_, linkItem) => <UrlInput linkItem={linkItem} />,
+    },
+    {
+      title: "Expiration Date",
+      key: "name",
+      dataIndex: "expirationDateTime",
+      render: (_, linkItem) => <ExpirationDate linkItem={linkItem} />,
+    },
+    {
+      title: "",
+      key: "action",
+      render: (_, linkItem) => (
+        <Tooltip title="Delete Link" placement="left">
+          <AsyncIconButton
+            onClick={() => deleteMutation.mutateAsync(linkItem.id)}
+            icon={<DeleteOutlined />}
+          />
+        </Tooltip>
+      ),
+    },
+  ];
+
+  return (
+    <div>
+      <div style={{ marginBottom: 8 }}>
+        You can create one or multiple Zarr link to this annotation below. The link can be used by
+        other tools to access the data in a streaming manner.
```

Comment:
```suggestion
      Create <a href="https://zarr.readthedocs.io/en/stable/">Zarr</a> streaming links to allow other tools to load the data of this annotation.
      Note that anyone with these links can access the data, regardless of other sharing settings.
```

Reply:
maybe this needs a target=_blank, just accidentally lost my wk tab :sweat_smile: